apis/pinned-posts/pinned-posts-manager.ts

```diff
@@ -0,0 +1,35 @@
+import db from "../../models/index";
+
+export class PinnedPostsManager {
+
+    public pinPost = async (publicKey: string, postHashHex: string) => {
+        const dbObject = {
+            publicKey,
+            postHashHex
+        };
+
+        const foundItem = await db.PinnedPosts.findOne({ where: { publicKey } });
+        if (!foundItem) {
+            await db.PinnedPosts.create(dbObject);
+        } else {
+            await db.PinnedPosts.update(dbObject, { where: { publicKey } });
+        }
+    }
+
+    public unpinPost = async (publicKey: string, postHashHex: string) => {
+        const deletedCount = await db.PinnedPosts.destroy(
+            {
+                where: {
+                    publicKey,
+                    postHashHex
```

Comment:
edge case?

If you pass a postHashHex that _isn't_ the currently pinned post then it's a no-op. Given that pinned is singular, should you "unpin a post" or just "remove your pinned post"?